pkg/server/server.go

```diff
@@ -1422,6 +1422,25 @@ If problems persist, please see ` + base.DocsURL("cluster-setup-troubleshooting.
 		s.cfg.AmbientCtx, s.recorder, DefaultMetricsSampleInterval, ts.Resolution10s, s.stopper,
 	)
 
+	log.Info(ctx, "starting graphite")
+	s.stopper.RunWorker(ctx, func(ctx context.Context) {
```

Comment:
If our configuration is based on a cluster setting, that implies that the push process can start (or stop) after the cockroach process.  That means you will need to change where your RunWorker loop is initialized - or perhaps as an alternative, that task will need to be "pauseable".